tools/snippet-testing/model/dependency.rb

```diff
@@ -10,33 +12,94 @@ class Dependency
     RUBY_NAME       = 'ruby'.freeze
     NODE_NAME       = 'node'.freeze
 
-    CS_V4 = '4.7.2'
-    CS_V5 = '5.5.0'
-
     AVAILABLE_LIBRARY_VERSION = {
-      CSHARP_NAME => ['4.x', '5.x'],
+      CSHARP_NAME => ['4.7.2', '5.x'],
       PHP_NAME    => ['4.10', '5.9.0'],
       PYTHON_NAME => ['5.6.0', '6.2.0'],
       RUBY_NAME   => ['4.13.0', '5.0.0.rc20'],
       NODE_NAME   => ['2.11.0', '3.0.0']
     }.freeze
 
-    CSHARP_DEPENDENCIES = {
-      AVAILABLE_LIBRARY_VERSION[CSHARP_NAME][0] => [
-        { name: 'Twilio', version: CS_V4 },
+    CSHARP_DEPENDENCIES = {}
+
+    def initialize
+      update_library_versions
+    end
+
+    def latest_version(versions)
+      versions = VersionSorter.sort(versions)
+      rc_or_alpha = /alpha1$|alpha\-1$|[0-9]{1,}a1$|rc[0-9]+$/
+      last_is_unstable = versions[-1] =~ rc_or_alpha
+
+      if versions[-2] =~ rc_or_alpha && !last_is_unstable
```

Comment:
@joliveros I'm a little concern about this approach (also, this line was hard for me to read).

Do you think is it possible to avoid fetching the latest `tag` and instead fetch by branch name? I was thinking that composer can do that `composer require twilio/sdk:dev-alpha` (which will fetch the `alpha` branch).
Because I believe the `alpha` branch already includes the latest stable and new changes and all new `rc` tags, but I'm not sure if I'm missing something in following this approach.